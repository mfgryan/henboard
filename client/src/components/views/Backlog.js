// react dep
import React from "react";
import { connect } from "react-redux";
import projects from "../../models/projects.js";
import lanes from "../../models/lanes.js";

// component dep
import Swimlane from "../containers/Swimlane";

// style dep
import { Col, Row } from "react-bootstrap";
import "../../css/Backlog.css";

export const Backlog = ({ project, lanes }) => {
    return (
        <Row className="board">
            {lanes.map((lane, index) =>
                <Col md={4} key={index}>
                    <Swimlane
                        project={project.project}
                        column={lane.column}
                    />
                </Col>
            )}
        </Row>
    );
};

const mapStateToProps = state => {
    let project = projects.getCurrentProject(state);
    let backlog = lanes.getBacklog(state, project.project);
    return {
        project: project,
        lanes: backlog
    };
};

const mapDispatchToProps = dispatch => {
    return {};
};

export default connect(mapStateToProps, mapDispatchToProps)(Backlog);
