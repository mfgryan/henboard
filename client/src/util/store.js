// redux dep
import { createStore, applyMiddleware } from "redux";
import { logger, validate, fn } from "../util/middleware.js"; // eslint-disable-line no-unused-vars
import rootReducer from "../reducers/reducers.js";

export default (function(){
    let store;
    return !store ? createStore(rootReducer, applyMiddleware(validate, logger, fn)) : store;
}());
