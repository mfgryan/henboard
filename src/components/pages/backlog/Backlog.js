import React, { Component } from "react";
import { Grid } from "react-bootstrap";
import Header from "../../header/Header";
import Board from "../../board/Board";

class Backlog extends Component {
    render() {
        const options = this.props.sprint.map((curSprint,index) => 
            ({val: curSprint.week, name: curSprint.week})
        );
        return (
            <Grid>
                <Header name={project.name} options={["backlog","project"]} />
                <Board lanes={project.backlog} />
            </Grid>
        );
    }
}
export default Backlog;
