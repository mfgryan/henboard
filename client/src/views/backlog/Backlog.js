// react dep
import React, { Component } from "react";

// component dep
import App from "../../components/App";

// style dep
import "./Backlog.css";

class Backlog extends Component {
    render() {
        return (
            <App>
                <p> Backlog here!</p>
            </App>
        );
    };
};

export default Backlog;
