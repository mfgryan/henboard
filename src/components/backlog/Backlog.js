import React, { Component } from "react";
import { Grid, Row, Col } from "react-bootstrap";
import "./Backlog.css";

class Backlog extends Component {
    render() {
        return (
            <div className='Backlog'>
            </div>
        );
    }
}
export default Backlog;
