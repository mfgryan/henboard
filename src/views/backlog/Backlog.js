// react dep
import React, { Component } from "react";

// style dep
import "./Backlog.css";

class Backlog extends Component {
    render() {
        return (
            <div className='Backlog'>
            </div>
        );
    };
};

export default Backlog;
