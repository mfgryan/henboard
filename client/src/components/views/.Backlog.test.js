import React from "react";
import Backlog from "./Backlog";
import renderer from "react-test-renderer";

// redux dep
import { Provider } from "react-redux";
import store from "../../util/store.js";

// router dep
import { BrowserRouter as Router } from "react-router-dom";

it("Backlog renders correctly", () => {
    const tree = renderer
        .create(
            <Router>
                <Provider store={store}>
                    <Backlog
                        project="henboard"
                        lanes={store.getState().lanes}
                    />
                </Provider>
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
