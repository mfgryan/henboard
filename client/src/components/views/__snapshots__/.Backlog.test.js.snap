exports[`test Backlog renders correctly 1`] = `
<div
  className="board row">
  <div
    className="col-md-4">
    <div
      className="swimlane">
      <div
        className="AddItem">
        <h2
          onClick={[Function]}>
          Backlog
        </h2>
        <div
          className="AddItemInput"
          style={
            Object {
              "visibility": "hidden",
            }
          }>
          <div
            className="form-group">
            <span
              className="input-group input-group-sm">
              <input
                className="form-control"
                id={undefined}
                onChange={[Function]}
                onKeyPress={[Function]}
                type="text"
                value="" />
              <span
                className="input-group-btn"
                type="text">
                <button
                  className="btn btn-default"
                  disabled={false}
                  onClick={[Function]}
                  type="button">
                  Add
                </button>
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        className="itemsArea"
        onDragOver={[Function]}
        onDrop={[Function]} />
    </div>
  </div>
</div>
`;
